controllers/workspace/devworkspace_controller.go

```diff
@@ -263,10 +237,7 @@ func (r *DevWorkspaceReconciler) Reconcile(req ctrl.Request) (reconcileResult ct
 			reqLogger.Info(provisionErr.Message)
 			return reconcile.Result{Requeue: true, RequeueAfter: provisionErr.RequeueAfter}, nil
 		case *metadata.ProvisioningError:
-			reqLogger.Info(fmt.Sprintf("DevWorkspace start failed: %s", provisionErr))
-			reconcileStatus.Phase = devworkspace.WorkspaceStatusFailed
-			reconcileStatus.Conditions[devworkspace.WorkspaceFailedStart] = fmt.Sprintf("Error provisioning metadata configmap: %s", provisionErr)
-			return reconcile.Result{}, nil
+			return r.failWorkspace(fmt.Sprintf("Error provisioning metadata configmap: %s", provisionErr), reqLogger, &reconcileStatus)
```

Comment:
Docs says updateWorkspaceStatus must be called https://github.com/devfile/devworkspace-operator/pull/345/files#diff-05e72230161a4d41c5a27fb453041c78839c193a9ddcb5066b1cb62c5a399529R343, or it's when you need to sync immediately but going to continue loop logic?

Reply:
I suppose the doc is unclear. I mean to say that `failWorkspace` doesn't do an update on its own. However, from this call site, the return value will be intercepted by the deferred function in the main reconcile loop, which will call `updateWorkspaceStatus`. I'm not sure the best way to explain this in a short doc though.

Reply:
Left a proposal for documentation